### PR TITLE
✨ NEW: Fence directives

### PR DESCRIPTION
**Pull request by rowanc1**
Now supports *not* parsing the internals of a directive, which is helpful for things like `math` directives.
Supported by the `skipParsing` option on the directive.

### CODE REVIEW

File: fixtures/directives.math.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Math directive:
+.
+```{math}
+:label: my_label
+w_{t+1} = (1 + r_{t+1}) s(w_t) + y_{t+1}
+```
+.
+<div class="math numbered" id="eq-my_label" number="1">
+w_{t+1} = (1 + r_{t+1}) s(w_t) + y_{t+1}
+</div>
```
Comment (rowanc1) [inline fixtures/directives.math.md:10]: Right now the math doesn't add `\[` around it. I am not sure we actually want this? The alternative is to use something else to notify mathjax.

File: src/directives/index.ts [hunk 2]
```diff
@@ -96,11 +142,20 @@ const numbering = (directives: Directives): RuleCore => (state) => {
...
+        const target = newTarget(state, name || label, directive.numbered);
+        token.meta.target = target;
+      }
+    }
+    if (token.type === DirectiveTokens.fence) {
+      const directive = getDirective(directives, token.attrGet('kind'));
+      const { name, label } = token.meta?.opts;
+      // Only number things if the directive supports numbering AND a name or label is provided
+      if (directive?.numbered && (name || label)) {
+        const target = newTarget(state, name || label, directive.numbered);
```
Comment (rowanc1) [inline src/directives/index.ts:158]: @choldgraf I am using `name` or `label` for things like equations a bit interchangably. The RST docs seem to say `name` is common, but EB suggests `label` for things like math? Let me know what the convention is!! This seems to work for now.

Comment (choldgraf) [reply src/directives/index.ts:158]: Hmmm - my personal understanding is that label is the "thing you reference" elsewhere (e.g. `myeq1`). E.g., the [rst quickstart](https://docutils.sourceforge.io/docs/user/rst/quickref.html) uses `label` quite a lot, very little "name". Where do you see `name` recommended in rst?

Comment (rowanc1) [reply src/directives/index.ts:158]: This is where I was going to:
https://docutils.sourceforge.io/docs/ref/rst/directives.html#math

Comment (rowanc1) [reply src/directives/index.ts:158]: The other place I have been looking is here:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#math

Which does say `label` -- missed that before.

Comment (rowanc1) [reply src/directives/index.ts:158]: image

Comment (rowanc1) [reply src/directives/index.ts:158]: Upon further conversation:

* Use name. Label is directive specific (e.g. equation).

Comment (rowanc1) [reply src/directives/index.ts:158]: See also https://github.com/sphinx-doc/sphinx/issues/8476
